core/trino-main/src/main/java/io/trino/node/AnnounceNodeAnnouncer.java

```diff
@@ -0,0 +1,174 @@
+    public AnnounceNodeAnnouncer(URI internalUri, Collection<URI> coordinatorUris, boolean coordinator, HttpClient httpClient)
+    {
+        URI currentUri = internalUri;
+        this.announceUris = coordinatorUris.stream()
+                .filter(not(currentUri::equals))
+                .map(uri -> uriBuilderFrom(uri).appendPath("/v1/announce").build())
+                .distinct()
+                .toList();
+        this.coordinator = coordinator;
+        this.httpClient = requireNonNull(httpClient, "httpClient is null");
+        this.currentHostAnnouncement = createStaticBodyGenerator(currentUri.toString(), UTF_8);
+    }
+
+    @Override
+    public void start()
+    {
+        if (announceUris.isEmpty()) {
+            if (!coordinator) {
+                log.warn("No coordinator URIs configured, skipping node state announcements");
+            }
+            return;
+        }
+
+        checkState(!announceExecutor.isShutdown(), "Announcer has been destroyed");
+        if (!started.compareAndSet(false, true)) {
+            return;
+        }
+
+        announceExecutor.scheduleWithFixedDelay(
+                () -> {
+                    try {
+                        forceAnnounce();
+                    }
+                    catch (RuntimeException e) {
+                        // this should not happen, but if it does we do not want to stop announcing
+                        log.error(e, "Error announcing node to coordinators");
+                    }
+                }, 5, 5, SECONDS);
+        forceAnnounce();
+    }
+
+    @Override
+    public ListenableFuture<?> forceAnnounce()
+    {
+        return Futures.allAsList(announceUris.stream()
+                .map(this::announce)
+                .toList());
+    }
+
+    private ListenableFuture<?> announce(URI announceUri)
+    {
+        Request request = preparePost()
+                .setUri(announceUri)
+                .setBodyGenerator(currentHostAnnouncement)
+                .setHeader(HttpHeaders.CONTENT_TYPE, TEXT_PLAIN)
+                .build();
+        ListenableFuture<StatusResponse> responseFuture = httpClient.executeAsync(request, createStatusResponseHandler());
+        Futures.addCallback(
+                responseFuture, new FutureCallback<>()
+                {
+                    @Override
+                    public void onSuccess(StatusResponse response)
+                    {
+                        int statusCode = response.getStatusCode();
+                        if (statusCode < 200 || statusCode >= 300) {
+                            logWarning("Failed to announce node state to %s: %s", announceUri, statusCode);
+                        }
+                    }
+
+                    @Override
+                    public void onFailure(Throwable t)
+                    {
+                        logWarning("Error announcing node state to %s: %s", announceUri, t.getMessage());
+                    }
+                }, directExecutor());
```

Comment:
FYI, this could be
```java
Futures.addCallback(addSuccessCallback(responseFuture, () -> {
    ...
});
Futures.addCallback(addExceptionCallback(responseFuture, t ->
    logWarning("Error announcing node state to %s: %s", announceUri, t.getMessage()));